service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/database/AzureDatabaseUtilsRunner.java

```diff
@@ -361,9 +373,36 @@ public void testDatabaseConnect(
   private void runAzureDatabaseUtils(
       AzureCloudContext azureCloudContext, UUID workspaceId, V1Pod podDefinition, String namespace)
       throws InterruptedException {
+    runAzureDatabaseUtils(
+        azureCloudContext, workspaceId, podDefinition, new HashMap<>(), namespace);
+  }
+
+  private void runAzureDatabaseUtils(
+      AzureCloudContext azureCloudContext,
+      UUID workspaceId,
+      V1Pod podDefinition,
+      Map<String, String> secretStringData,
+      String namespace)
+      throws InterruptedException {
     var aksApi = kubernetesClientProvider.createCoreApiClient(azureCloudContext, workspaceId);
+
     // strip underscores to avoid violating azure's naming conventions for pods
     var safePodName = podDefinition.getMetadata().getName();
+    var doSecretCleanup = false;
+    if (getPodDefinitionEnvVarsWithSecretRefs(podDefinition).size() > 0
+        && secretStringData.isEmpty()) {
+      throw new RuntimeException(
+          "definition of pod {} contains env vars that refer to secrets, but no secret data was provided."
+              .formatted(safePodName));
+    } else {
+      try {
+        createSecret(aksApi, secretStringData, safePodName, namespace); // name secret after pod
+        doSecretCleanup = true;
```

Comment:
I was thinking that this would prevent us from trying to delete a secret that doesn't exist (in the `finally` block below). but I suppose the worst that will happen is a logged warning that the secret doesn't exist.